engine/pom.xml

```diff
@@ -104,6 +100,13 @@
             <version>${cloudwatch.version}</version>
         </dependency>
 
+        <!-- Retries -->
+        <dependency>
+            <groupId>dev.failsafe</groupId>
```

Comment:
why don't we use resilience4j? The api is much cleaner imo and even Netflix deprecated their hystrix project and uses resilience4j instead.

Reply:
I added failsafe here because I've used that lib in many projects for years and it never caused any production issue. I don't really know resilience4j so I don't have a preference.

Which part of the failsafe API used in this PR isn't clean enough?

Reply:
Just took a look at it. Version 3 of it looks much better than the previous one. I remember we updated from failsafe 1 to 2 some years ago and got a generic api which worked with Object instead of concrete classes. But nevermind, version 3 looks good. We can stay with failsafe. In general they are doing the same.